### ISSUE TITLE
Incorrect parameters passed into pods

### ISSUE THREAD

**Issue by warunapww** [argoproj/argo#1738]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT

**What happened**:
I was running the following workflow

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: dep-sample-
spec:
  entrypoint: dep-sample

  arguments:
    parameters:
    - name: msg-list
      value: |
        ["A","B","C"]

  templates:
  - name: echo
    inputs:
      parameters:
      - name: message
    container:
      image: alpine:3.7
      command: [echo, "{{inputs.parameters.message}}"]
  - name: dep-sample-C
    inputs:
      parameters:
      - name: msg
    steps:
    - - name: T1
        template: echo
        arguments:
          parameters: [{name: message, value: "{{inputs.parameters.msg}}"}]
    - - name: T2         #T2 depends on T1
        template: echo
        arguments:
          parameters: [{name: message, value: "{{inputs.parameters.msg}}"}]

  - name: dep-sample
    inputs:
      parameters:
      - name: msg-list
    dag:
      tasks:
      - name: PT
        template: dep-sample-C
        arguments:
          parameters: [{name: msg, value: "{{item}}"}]
        withParam: "{{inputs.parameters.msg-list}}"
```

```
argo get  dep-sample-6sp8k

Parameters:
  msg-list:          ["A","B","C"]


STEP                              PODNAME                      DURATION  MESSAGE
 ✔ dep-sample-6sp8k (dep-sample)
 ├-✔ PT(0:A) (dep-sample-C)
 | ├---✔ T1 (echo)                dep-sample-6sp8k-1828940143  1s
 | └---✔ T2 (echo)                dep-sample-6sp8k-3132179825  1s
 ├-✔ PT(1:B) (dep-sample-C)
 | ├---✔ T1 (echo)                dep-sample-6sp8k-1070029273  1s
 | └---✔ T2 (echo)                dep-sample-6sp8k-1964569871  1s
 └-✔ PT(2:C) (dep-sample-C)
   ├---✔ T1 (echo)                dep-sample-6sp8k-2342097303  1s
   └---✔ T2 (echo)                dep-sample-6sp8k-791003929   1s
```

The value of ‘message’ passed to T2 of PT(1:B) and PT(2:C) is incorrect. It passes value 'A' instead of 'B' and 'C' respectively.

**What you expected to happen**:
The value of ‘message’ passed to T2 of PT(1:B) to be 'B'
The value of ‘message’ passed to T2 of PT(1:C) to be 'C'

**How to reproduce it (as minimally and precisely as possible)**:
Save the above workflow as `dep-sample.yaml`
then run `argo submit --watch dep-sample.yaml`

Check the actual parameter values in the output of the following command
`kubectl get wf dep-sample-xxx -o YAML > dep-sample-xxx.yaml`

**Anything else we need to know?**:
I believe this issue is related to #1713  and #1725

**Environment**:
- Argo version:
```
$ argo version
```

argo: v2.4.2
  BuildDate: 2019-10-21T18:39:29Z
  GitCommit: 675c66267f0c916de0f233d8101aa0646acb46d4
  GitTreeState: clean
  GitTag: v2.4.2
  GoVersion: go1.11.5
  Compiler: gc
  Platform: linux/amd64

- Kubernetes version :
```
$ kubectl version -o yaml
```
clientVersion:
  buildDate: "2019-10-15T19:18:23Z"
  compiler: gc
  gitCommit: c97fe5036ef3df2967d086711e6c0c405941e14b
  gitTreeState: clean
  gitVersion: v1.16.2
  goVersion: go1.12.10
  major: "1"
  minor: "16"
  platform: linux/amd64
serverVersion:
  buildDate: "2019-10-15T19:09:08Z"
  compiler: gc
  gitCommit: c97fe5036ef3df2967d086711e6c0c405941e14b
  gitTreeState: clean
  gitVersion: v1.16.2
  goVersion: go1.12.10
  major: "1"
  minor: "16"
  platform: linux/amd64

**Other debugging information (if applicable)**:
- workflow result:
```
$ argo get <workflowname>
```
Name:                dep-sample-6sp8k
Namespace:           testns
ServiceAccount:      default
Status:              Succeeded
Created:             Tue Nov 05 12:45:00 -0500 (34 minutes ago)
Started:             Tue Nov 05 12:45:00 -0500 (34 minutes ago)
Finished:            Tue Nov 05 12:45:05 -0500 (34 minutes ago)
Duration:            5 seconds
Parameters:
  msg-list:          ["A","B","C"]

STEP                              PODNAME                      DURATION  MESSAGE
 ✔ dep-sample-6sp8k (dep-sample)
 ├-✔ PT(0:A) (dep-sample-C)
 | ├---✔ T1 (echo)                dep-sample-6sp8k-1828940143  1s
 | └---✔ T2 (echo)                dep-sample-6sp8k-3132179825  1s
 ├-✔ PT(1:B) (dep-sample-C)
 | ├---✔ T1 (echo)                dep-sample-6sp8k-1070029273  1s
 | └---✔ T2 (echo)                dep-sample-6sp8k-1964569871  1s
 └-✔ PT(2:C) (dep-sample-C)
   ├---✔ T1 (echo)                dep-sample-6sp8k-2342097303  1s
   └---✔ T2 (echo)                dep-sample-6sp8k-791003929   1s

- executor logs:
```
$ kubectl logs <failedpodname> -c init
$ kubectl logs <failedpodname> -c wait
```
- workflow-controller logs:
```
$ kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name)
```

**Comment by simster7** [comment]
Duplicate of https://github.com/argoproj/argo/issues/1713. A fix is being worked on currently.

**Comment by simster7** [comment]
Hello @warunapww, [`v2.4.3`](https://github.com/argoproj/argo/releases/tag/v2.4.3) was just released with a fix for this issue, please take a look! Thank you for your patience.
